alethzero/MainFace.cpp

```diff
@@ -54,6 +74,18 @@ void Plugin::addAction(QAction* _a)
 	m_main->addAction(_a);
 }
 
+QAction* Plugin::addMenuItem(QString _n, QString _menuName, bool _sep)
+{
+	QAction* a = new QAction(_n, main());
+	QMenu* m = main()->findChild<QMenu*>(_menuName);
+	if (_sep)
+		m->addSeparator();
+	m->addAction(a);
+	return a;
+}
+
+std::vector<std::function<Plugin*(MainFace*)>>* MainFace::s_linkedPlugins = nullptr;
```

Comment:
Also, I'm not sure this will work properly on all platforms.  It may happen that static initialization will happen after plugin registration, zeroing the pointer. I guess this does not happen because the compiler optimizes the zero static initialization to no-op since static memory is zeroed before any code is called anyway. Still better not rely on this and leave the variable uninitialized.